src/typechecker/FStar.TypeChecker.Normalize.fs

```diff
@@ -663,9 +664,9 @@ let reduce_primops norm_cb cfg env tm =
          | _ -> tm
    end
 
-let reduce_equality norm_cb cfg tm =
+let reduce_equality norm_cb cfg env tm =
     reduce_primops norm_cb ({cfg with steps = { default_steps with primops = true };
-                              primitive_steps=equality_ops}) tm
+                              primitive_steps=equality_ops}) env tm
```

Comment:
Ha, took me a second to realize how this change could possibly be type correct :). Thanks for fixing it!

What do you think about adding a ` : term` to the let so this doesn't happen again?

Reply:
Where exactly in the let?

Reply:
I meant annotating like this:
```fstar
let reduce_equality norm_cb cfg env tm : term =
```
so if a new argument is added, this will break and we need to explicitly take it in here. Anyway no big deal, I can also add it later.